src/wallet/api.rs

```diff
@@ -1060,48 +1074,177 @@ impl Wallet {
-        // Get UTXOs from the descriptor and swap coin methods.
-        let mut seed_coin_utxo = self.list_descriptor_utxo_spend_info()?;
+        // TODO : Create a user input variable with the boarder merge split refactor
+        let num_outputs = 1; // Number of outputs
+
+        // Get spendable UTXOs (regular coins and swap coins)
+        let seed_coin_utxo = self.list_descriptor_utxo_spend_info()?;
         let mut swap_coin_utxo = self.list_incoming_swap_coin_utxo_spend_info()?;
-        seed_coin_utxo.append(&mut swap_coin_utxo);
 
         // Retrieve currently locked UTXOs via RPC.
         // Assume that self.rpc.listlockunspent() returns a Vec<OutPoint>.
         let locked_utxos: Vec<OutPoint> = self.list_lock_unspent()?;
 
-        // Filter out UTXOs that are already locked and exclude fidelity coins.
-        let mut unspents = seed_coin_utxo
+        // Combine seed and swap coins
+        let mut seed_swap_coins = seed_coin_utxo;
+        seed_swap_coins.append(&mut swap_coin_utxo);
+
+        // Filter out locked UTXOs
+        let unspents = seed_swap_coins
             .into_iter()
-            .filter(|(utxo, spend_info)| {
+            .filter(|(utxo, _)| {
                 let outpoint = OutPoint::new(utxo.txid, utxo.vout);
                 !locked_utxos.contains(&outpoint)
-                    && !matches!(spend_info, UTXOSpendInfo::FidelityBondCoin { .. })
             })
             .collect::<Vec<_>>();
 
-        // Sort in descending order so that we use the largest UTXOs first.
-        unspents.sort_by(|a, b| b.0.amount.cmp(&a.0.amount));
+        // Assertion statement : fidelity coins are not included
+        assert!(
+            unspents.iter().all(|(_, spend_info)| !matches!(
+                spend_info,
+                UTXOSpendInfo::FidelityBondCoin { .. }
+            )),
+            "Fidelity coins are not included in coin selection"
+        );
 
-        let mut selected_utxo = Vec::new();
-        let mut remaining = amount;
+        // Filter out UTXOs that are already locked and exclude fidelity coins.
+        // let mut unspents = seed_swap_coins
+        //     .into_iter()
+        //     .filter(|(utxo, spend_info)| {
+        //         let outpoint = OutPoint::new(utxo.txid, utxo.vout);
+        //         !locked_utxos.contains(&outpoint)
+        //             && !matches!(spend_info, UTXOSpendInfo::FidelityBondCoin { .. })
+        //     })
+        //     .collect::<Vec<_>>();
+
+        // Get next external and internal addresses using dedicated wallet APIs
+        let external_address = self.get_next_external_address()?;
+        let internal_address = self.get_next_internal_addresses(1)?[0].clone();
+
+        let external_address_script_pubkey = ScriptBuf::new_p2pkh(
+            &external_address
+                .pubkey_hash()
+                .ok_or_else(|| WalletError::General("Could not get pubkey hash".to_string()))?,
+        );
```

Comment:
This will cause an error if the external address is anything other than p2pkh. That should not happen.

For now its okay. Mark a TODO here to tackle this later.